3.5-windowsservercore-1709/runtime/Dockerfile

```diff
@@ -1,11 +1,37 @@
+# escape=`
+
 FROM microsoft/windowsservercore:1709
 
-ADD install /build
-RUN DISM /Online /Add-Package /PackagePath:C:\build\microsoft-windows-netfx3-ondemand-package.cab & \
-    DISM /Online /Add-Package /PackagePath:C:\build\patch\Windows10.0-KB3213986-x64.cab  & \
-    rd /s /q C:\build
-RUN set COMPLUS_NGenProtectedProcess_FeatureEnabled=0 & \
-    \Windows\Microsoft.NET\Framework64\v4.0.30319\ngen update & \
-    \Windows\Microsoft.NET\Framework\v4.0.30319\ngen update & \
-    \Windows\Microsoft.NET\Framework64\v2.0.50727\ngen update & \
+# Install .NET Fx 3.5
+RUN powershell -Command `
+        $ErrorActionPreference = 'Stop'; `
+        $ProgressPreference = 'SilentlyContinue'; `
+        Invoke-WebRequest `
+            -UseBasicParsing `
+            -Uri "https://dotnetbinaries.blob.core.windows.net/dockerassets/microsoft-windows-netfx3-1709.zip" `
+            -OutFile microsoft-windows-netfx3.zip; `
+        Expand-Archive microsoft-windows-netfx3.zip; `
+        Remove-Item -Force microsoft-windows-netfx3.zip; `
+        Add-WindowsPackage -Online -PackagePath .\microsoft-windows-netfx3\microsoft-windows-netfx3-ondemand-package.cab; `
+        Remove-Item -Force -Recurse microsoft-windows-netfx3
+
+# Apply latest patch
+RUN powershell -Command `
+        $ErrorActionPreference = 'Stop'; `
+        $ProgressPreference = 'SilentlyContinue'; `
+        Invoke-WebRequest `
+            -UseBasicParsing `
+            -Uri "http://download.windowsupdate.com/d/msdownload/update/software/secu/2017/11/windows10.0-kb4048955-x64_f214597fd57cd8284a363ada825a3d8ca16bf632.msu" `
```

Comment:
The Uri should be a parameter. When building candidate images we point to a local file share instead

Reply:
I had thought about this but how would you get the credentials to access the file share within the container?  The build container is not domain joined and is running as the ContainerAdministrator.

Reply:
Off the top of my head, I don't if docker build args support default values, but I think that would be the best case for us.

Reply:
Dockerfile ARGS support default values.  The part I am missing is that is you pass in a local file share url at build time, how is the build container going to get access to it?

Reply:
I don't have a good answer 😕
@mcy94w Thoughts?